src/components/tools/Partials/ChangeLocationStation/ChangeLocationStation.svelte

```diff
@@ -0,0 +1,155 @@
+<script>
+  import { createEventDispatcher } from "svelte";
+  import { Modal } from "carbon-components-svelte";
+
+  import SelectBoundary from "./Boundary.svelte";
+  import LocationMap from "./Map.svelte";
+  import Search from "./Search.svelte";
+
+  /** @type {!Object} the locationStore's location object */
+  export let location;
+
+  /** @type {?Object} the locationStore's boundary object */
+  export let boundary;
+
+  /** @type {?Array} specify the list of boundaries to be displayed in the dropdown */
+  export let boundaryList;
+
+  /** @type {?Object} specify which station layer to use. Overrides boundaryList */
+  export let stationsLayer;
+
+  /** @type {!boolean} specify whether the CA state boundary should be included in the boundaryList */
+  export let addStateBoundary = false;
+
+  /** @type {!boolean} specify whether a user may upload custom boundary data */
+  export let enableUpload = false;
+
+  /** @type {!boolean} specify whether or not the modal is shown */
+  export let open = false;
+
+  const dispatch = createEventDispatcher();
+
+  let isStationSelector = Boolean(stationsLayer);
+
+  let helpText = isStationSelector
+    ? `Select a station on the map or enter an address in the search box to 
+        select the nearest station.`
+    : `Click on the map or enter an address in the search box. To explore data
+        for a larger extent (e.g. county), select a boundary first.`;
+
+  let headingTitleText = isStationSelector
+    ? "Change Station"
+    : "Change Location";
+
+  let currentLocation = location;
+  let currentBoundary = boundary;
+
+  let searchPlaceholder = isStationSelector
+    ? "Enter place name or address"
+    : boundary && boundary.metadata
+    ? updatePlaceholderText(boundary.metadata.placeholder)
+    : "";
+
+  function updatePlaceholderText(boundary) {
+    if (boundary && boundary.metadata) {
+      searchPlaceholder = `Enter ${boundary.metadata.placeholder}`;
+    }
+  }
+
+  async function handleSubmit() {
+    if (
+      (isStationSelector && currentLocation) ||
+      (!isStationSelector && currentBoundary && currentLocation)
+    ) {
+      open = false;
+      dispatch("change", {
+        ...(currentBoundary && { boundaryId: currentBoundary.id }),
+        location: currentLocation,
+      });
+    }
+  }
+
+  function handleSearchSelect({ detail }) {
+    currentLocation = detail;
+  }
+
+  function handleChangeBoundary({ detail }) {
+    currentBoundary = detail;
+    updatePlaceholderText(currentBoundary);
```

Comment:
Probably won't need this if the handling of placeholder text is moved to `Search.svelte`